solidity/contracts/ECDSARewards.sol

```diff
@@ -266,6 +294,32 @@ contract ECDSARewards is Rewards {
         }
     }
 
+    function calculateLifespanReward(bytes32 _keep, uint256 amount)
+        internal returns (uint256) {
+        uint256 createdAt = _getCreationTime(_keep);
+
+        Heartbeat lastHeartbeat = keepHeartbeats[toAddress(_keep)];
+
+        uint256 lastTimestamp = uint256(lastHeartbeat.timestamp);
+        uint256 keepBond = uint256(lastHeartbeat.bondAmount);
+
+        uint256 rewardLifespan = lastTimestamp == 0
+            ? assumedMinimumLifetime
```

Comment:
In this setup, do I need to send a heartbeat before the funding timeout? My understanding is that I don't have to and the only heartbeat needed is once the redemption has been requested.